### DISCUSSION TITLE
Shared cross-network edges in EpiModel

### DISCUSSION THREAD

**Discussion by AdrienLeGuillou** [EpiModel/EpiModel#692] [EpidemicMethods]
@smjenness @sgoodreau @chad-klumb @martinamorris

As we discussed on Friday, I checked how often a dyad had an edge in multiple network simultaneously.

I ran the BigNet EpiModel HIV simulation over 10 years (520 weeks) with 10k nodes per network.

Overall, there was 7k edges in the main network, 16k in casual and 93k in the instantaneous one.

In terms of concurrency, there were 3 cases of main and instantaneous and 13 of casual and instantaneous. So 16 dyads and 32 edges.

**Comment by sgoodreau** [comment]
Thanks @AdrienLeGuillou!

In terms of impact on our epidemic models, 16/116k seems non-significant, so that is great news.  And 10k nodes is the smallest we ever go, and the problem should be even less in larger networks with the same mean degree.

But of course it remains an inelegant aspect of the modeling approach, regardless of impact.

**Comment by sgoodreau** [comment]
Just changed the name of the issue since "concurrent" has such a specific and long-standing meaning in our work.

**Comment by AdrienLeGuillou** [comment]
Good catch, that was a poor choice indeed

**Comment by martinamorris** [comment]
Thanks @AdrienLeGuillou for the test.  

It would be helpful to make the code for that available.  Do we have a mechanism for something like this?  It's almost like a network dx...

**Comment by AdrienLeGuillou** [comment]
@martinamorris  For now the code I use is here:
 https://github.com/EpiModel/BigNets/blob/pdf_scstep/R/09-test_netsim.R

in the last section of the file.

I am using the cumulative edgelist that I implemented some time ago. I have to process it "manually" as I did not make the helper functions to work with a `netsim` object, only with `dat` inside the modules.

`el_cuml_df` is a data.frame with the following columns: head, tail, start, stop, network
grouping by `head` and `tail` I could check the shared cross-network edges

**Comment by smjenness** [comment]
@AdrienLeGuillou : that is good to hear! Can you take the same data generated for this analysis and estimate the number of times an edge spell recurs with the same dyad within a network type? This was the other form of potential bias we discussed during that call. e.g., main partnership edge between nodes 48 and 150 occurs at in time interval [1-100] and [200-300]. Thanks!

**Comment by AdrienLeGuillou** [comment]
@smjenness : strangely this almost never happens

```
el_cuml_df %>%
  ungroup() %>%
  group_by(network, head, tail) %>%
  summarize(n = n()) %>%
  ungroup() %>%
  group_by(network) %>%
  summarize(repeated = sum(n > 1))

> # A tibble: 3 × 2
>   network repeated
>     <int>    <int>
> 1       1        0
> 2       2        0
> 3       3        0
```
Here it shows that no edge is repeated on a network over the 10 years of simulation.
in reality it occurred 16 times on the one-of network where the edges were repeated on two consecutive step:

```
el_cuml_df %>%
  filter(network == 3, stop - start > 0) %>%
  mutate(duration = stop - start + 1) %>%
  print(n = 200)

> # A tibble: 16 × 6
>     head  tail start  stop network duration
>    <dbl> <dbl> <dbl> <dbl>   <int>    <dbl>
>  1  3718  7528     9    10       3        2
>  2  3400  5385    38    39       3        2
>  3  5509  6011    68    69       3        2
>  4  5915  9267   128   129       3        2
>  5   864  8278   170   171       3        2
>  6   666  6773   185   186       3        2
>  7  8898  9430   192   193       3        2
>  8  6886  8285   256   257       3        2
>  9  1934  4681   292   293       3        2
> 10  1318  3718   293   294       3        2
> 11  3894  5000   389   390       3        2
> 12 11135 11686   401   402       3        2
> 13  1192  5090   416   417       3        2
> 14  2061  2713   478   479       3        2
> 15   316  9489   479   480       3        2
> 16  5342  8911   508   509       3        2
```

Notes: 
- the way `el_cuml` calculates the edgelist does not permit the distinction between an edge of length 2 and two consecutive edges of length one. I just know that the edges on the network 3 only exist for 1 step so when they appear to be of length 2 I know they are repeated.
- On the `el_cuml`, an edge is active from `start` to `end`, both inclusive. That's why duration is `stop - start + 1`
